src/Metadata/PropertyTypeClass.php

```diff
@@ -71,6 +71,9 @@ public function merge(PropertyType $other): PropertyType
         if (is_a($this->getClassName(), Collection::class, true) && (($other instanceof PropertyTypeIterable) && $other->isCollection())) {
             return $other->merge($this);
         }
+        if (is_a($this->getClassName(), \DateTimeInterface::class, true) && ($other instanceof PropertyTypeDateTime)) {
```

Comment:
is this needed? if i read the code correctly, the parser will start to create a PropertyDateTime for DateTimeInterface as well, no?